src/library_wasi.js

```diff
@@ -175,22 +175,25 @@ var WasiLibrary = {
 #endif
   fd_write__sig: 'iiiii',
   fd_write: function(fd, iov, iovcnt, pnum) {
+    iov = Number(iov);
+    iovcnt = Number(iovcnt);
+    pnum = Number(pnum);
 #if SYSCALLS_REQUIRE_FILESYSTEM
     var stream = SYSCALLS.getStreamFromFD(fd);
     var num = SYSCALLS.doWritev(stream, iov, iovcnt);
 #else
     // hack to support printf in SYSCALLS_REQUIRE_FILESYSTEM=0
     var num = 0;
     for (var i = 0; i < iovcnt; i++) {
-      var ptr = {{{ makeGetValue('iov', 'i*8', 'i32') }}};
-      var len = {{{ makeGetValue('iov', 'i*8 + 4', 'i32') }}};
+      var ptr = {{{ makeGetValue('iov', `i*${Runtime.POINTER_SIZE}*2`, `${Runtime.POINTER_TYPE}`) }}};
+      var len = {{{ makeGetValue('iov', `i*${Runtime.POINTER_SIZE}*2 + ${Runtime.POINTER_SIZE}`, `${Runtime.POINTER_TYPE}`) }}};
       for (var j = 0; j < len; j++) {
         SYSCALLS.printChar(fd, HEAPU8[ptr+j]);
       }
       num += len;
     }
 #endif // SYSCALLS_REQUIRE_FILESYSTEM
-    {{{ makeSetValue('pnum', 0, 'num', 'i32') }}}
```

Comment:
That would have to be `PTR_TO_SIZE_T_TYPE` then? And still produce `i64` like this one does? just for clarity? and do that for every pointer type we come across?

Reply:
It just seems a bit strange because pnum is an integer right? not a pointer?  The thing being stored here is not point pointer but a quantity, right?

I think it would work to just have `${Runtime.SIZE_TYPE}` here?    Am I misunderstanding?

Reply:
Sorry, you are right, the type is `size_t`, not `size_t *` (the type of `pnum`), will add another constant for clarity